src/gateway/router/routes/deployContractRoute.ts

```diff
@@ -124,29 +150,50 @@ function tagValue(name: string, tags: GQLTagInterface[]): string | undefined {
   return tag?.value;
 }
 
-function prepareTags(transaction: Transaction, originalAddress: string): {tags: GQLTagInterface[], testnet: string | null} {
+async function prepareTags(
+  transaction: Transaction,
+  originalOwner: string,
+  logger: any,
+  arweave: Arweave
+): Promise<{ tags: GQLTagInterface[]; testnet: string | null; originalAddress: string; isEvmSigner: boolean }> {
   const decodedTags: GQLTagInterface[] = [];
-  let testnet = null;
+  let testnet = null,
+    isEvmSigner = false,
+    originalAddress = '';
 
   transaction.tags.forEach((tag) => {
-    const key = tag.get('name', {decode: true, string: true});
-    const value = tag.get('value', {decode: true, string: true});
+    const key = tag.get('name', { decode: true, string: true });
+    const value = tag.get('value', { decode: true, string: true });
     decodedTags.push({
       name: key,
       value: value,
     });
     if (key == 'Warp-Testnet') {
       testnet = value;
     }
+    if (key == 'Signature-Type' && value == 'ethereum') {
+      logger.info(`Signature type for ${transaction.id}`, value);
+      originalAddress = originalOwner;
+      isEvmSigner = true;
+    }
   });
 
+  if (!isEvmSigner) {
+    originalAddress = await arweave.wallets.ownerToAddress(originalOwner);
+  }
+
   const tags = [
-    {name: 'Uploader', value: 'RedStone'},
-    {name: 'Uploader-Contract-Owner', value: originalAddress},
-    {name: 'Uploader-Tx-Id', value: transaction.id},
-    {name: 'Uploader-Bundler', value: BUNDLR_NODE2_URL},
+    { name: 'Uploader', value: 'RedStone' },
+    { name: 'Uploader-Contract-Owner', value: originalAddress },
+    { name: 'Uploader-Tx-Id', value: transaction.id },
+    { name: 'Uploader-Bundler', value: BUNDLR_NODE2_URL },
     ...decodedTags,
   ];
 
-  return { tags, testnet };
+  return { tags, testnet, originalAddress, isEvmSigner };
+}
+
+async function verifyEvmSignature(tx: Transaction, logger: any) {
+  const signatureVerification = new EvmSignatureVerificationServerPlugin();
```

Comment:
do we need to create the plugin instance each time? does it store some state?

Reply:
done.